src/main/kotlin/com/chattriggers/ctjs/minecraft/listeners/ClientListener.kt

```diff
@@ -102,6 +107,31 @@ object ClientListener {
         this.draggedState[button] = State(Client.getMouseX(), Client.getMouseY())
     }
 
+    @SubscribeEvent
+    fun onNetworkEvent(event: FMLNetworkEvent.ClientConnectedToServerEvent) {
+        event.manager.channel().pipeline().addAfter("fml:packet_handler", "CT_packet_handler", object : ChannelDuplexHandler() {
+            override fun channelRead(ctx: ChannelHandlerContext?, msg: Any?) {
+                val packetReceivedEvent = CancellableEvent()
+
+                if (msg is Packet<*>)
+                    TriggerType.PACKET_RECEIVED.triggerAll(msg, packetReceivedEvent)
+
+                if (!packetReceivedEvent.isCanceled())
+                    ctx?.fireChannelRead(msg)
+            }
+
+            override fun write(ctx: ChannelHandlerContext?, msg: Any?, promise: ChannelPromise?) {
```

Comment:
I believe this should be `channelWrite` and not `write`. While it may not make a difference, it makes the most sense to use 2 methods to match as opposed to 2 that don't (the `ctx?.write` would also need to be changed to `ctx?.fireChannelWrite`).

Reply:
I agree the naming is rather strange but there isn't any `channelWrite` method so blame netty not me.

Reply:
Seems like that is true, I swear there was a `channelWrite` but its either a newer netty version or I am just misremembering.

Reply:
I checked netty docs and there doesn't seem to be a `channelWrite` in any of the versions. Correct me if I'm wrong. I don't think that changed in newer versions.